moveit_planners/ompl/ompl_interface/src/generate_state_database.cpp

```diff
@@ -0,0 +1,140 @@
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Authors: Ioan Sucan, Michael Goerner */
+
+#include <tf/transform_datatypes.h>
+#include <moveit/ompl_interface/ompl_interface.h>
+#include <moveit/robot_model_loader/robot_model_loader.h>
+#include <moveit/profiler/profiler.h>
+
+#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+
+#include <moveit/kinematic_constraints/utils.h>
+
+#include <boost/math/constants/constants.hpp>
+#include <sstream>
+
+static const std::string ROBOT_DESCRIPTION = "robot_description";
+
+static const std::string CONSTRAINT_PARAMETER = "constraints";
+
+void computeDB(const planning_scene::PlanningScenePtr& scene, moveit_msgs::Constraints& constraints,
+               ros::NodeHandle& nh)
+{
+  // required by addConstraintApproximation
+  scene->getCurrentStateNonConst().update();
+
+  ompl_interface::OMPLInterface ompl_interface(scene->getRobotModel());
+
+  ompl_interface::ConstraintApproximationConstructionOptions opt;
+
+  // number of states in joint space approximation
+  opt.samples = nh.param<int>("state_cnt", 10000);
+
+  // useful default parameterization
+  // - disable edges by default
+  // - local planning in JointModel state space
+  opt.edges_per_sample = nh.param<int>("edges_per_sample", 0);
+  opt.max_edge_length = nh.param<double>("max_edge_length", 0.2);
+  opt.explicit_motions = nh.param<bool>("explicit_motions", true);
+  opt.explicit_points_resolution = nh.param<double>("explicit_points_resolution", 0.05);
+  opt.max_explicit_points = nh.param<int>("max_explicit_points", 200);
+  opt.state_space_parameterization = nh.param<std::string>("state_space_parameterization", "JointModel");
+
+  ROS_INFO_STREAM("Generating Joint Space Constraint Approximation Database for constraint:\n" << constraints);
+
+  const std::string planning_group = nh.param<std::string>("planning_group", "<unspecified planning group>");
```

Comment:
Is this actually a good default? Maybe return with an error msg would be better here